services/graphql-server/src/graphql/definitions/email/campaign.js

```diff
@@ -1,11 +1,80 @@
 const gql = require('graphql-tag');
 
 module.exports = gql`
+  extend type Query {
+    emailCampaign(input: EmailCampaignQueryInput!): EmailCampaign
+      @findOne(model: "email.Campaign", using: { id: "_id" })
+    emailCampaigns(input: EmailCampaignQueryInput!): EmailCampaignConnection
+  }
 
-type EmailCampaignTestRecipient {
-  firstName: String
-  lastName: String
-  email: String
-}
+  input EmailCampaignQueryInput {
+    status: ModelStatus = active
+    id: ObjectID
+    productId: ObjectID
+    sort: EmailCampaignsSortInput = {}
+    pagination: PaginationInput = {}
+    beginning: CampaignBeginningInput = {}
+    ending: CampaignEndingInput = {}
+  }
+
+  input CampaignBeginningInput {
+    before: Date
+    after: Date
+  }
+
+  input CampaignEndingInput {
+    before: Date
+    after: Date
+  }
+
+  enum EmailCampaignsSortField {
+    status
+    deploymentDate
+    scheduled
```

Comment:
Have indexes been created for sorting on all of these fields?

Reply:
Not yet but I know I would need them.